### ISSUE TITLE
[BUG] async-tungstenite missing features

### ISSUE THREAD

**Issue by Adrien-Bodineau** [jgraef/reqwest-websocket#32]
Using the `0.28.1` of async-tungstenite this crate do not compile anymore 

```
error[E0599]: the method `poll_ready_unpin` exists for struct `WebSocketStream<Compat<Upgraded>>`, but its trait bounds were not satisfied                                                                                                                                                                         
   --> C:\Users\XXX\.cargo\registry\src\index.crates.io-6f17d22bba15001f\reqwest-websocket-0.4.3\src/lib.rs:301:20
    |
301 |         self.inner.poll_ready_unpin(cx).map_err(Into::into)
    |                    ^^^^^^^^^^^^^^^^
    |
   ::: C:\Users\XXX\.cargo\registry\src\index.crates.io-6f17d22bba15001f\async-tungstenite-0.28.1\src/lib.rs:227:1
    |
227 | pub struct WebSocketStream<S> {
    | ----------------------------- doesn't satisfy `_: Sink<_>` or `_: SinkExt<_>`
    |
    = note: the following trait bounds were not satisfied:
            `async_tungstenite::WebSocketStream<tokio_util::compat::Compat<Upgraded>>: futures_util::Sink<_>`
            which is required by `async_tungstenite::WebSocketStream<tokio_util::compat::Compat<Upgraded>>: SinkExt<_>`
```

this is because of the `default-features = false` in `Cargo.toml` therefore it lacks the new `async-tungstenite` default feature `futures-03-sink`

**Comment by Adrien-Bodineau** [comment]
or remove the `default-features = false` and yank the bad version

**Comment by jgraef** [comment]
Any clues on why CI didn't catch that? I (obviously) also compiled after changing the `async-tungstenite` dependency and didn't get any errors.

Now I did a `cargo update` and I get the following error:

```
    Checking async-tungstenite v0.28.1
error[E0432]: unresolved imports `futures_task::waker_ref`, `futures_task::ArcWake`
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-tungstenite-0.28.1/src/compat.rs:9:20
   |
9  | use futures_task::{waker_ref, ArcWake};
   |                    ^^^^^^^^^  ^^^^^^^ no `ArcWake` in the root
   |                    |
   |                    no `waker_ref` in the root
   |
note: found an item that was configured out
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:37:5
   |
37 | mod waker_ref;
   |     ^^^^^^^^^
note: the item is gated behind the `alloc` feature
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:36:7
   |
36 | #[cfg(feature = "alloc")]
   |       ^^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:40:28
   |
40 | pub use crate::waker_ref::{waker_ref, WakerRef};
   |                            ^^^^^^^^^
note: the item is gated behind the `alloc` feature
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:39:7
   |
39 | #[cfg(feature = "alloc")]
   |       ^^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:26:26
   |
26 | pub use crate::arc_wake::ArcWake;
   |                          ^^^^^^^
note: the item is gated behind the `alloc` feature
  --> /home/emma/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-task-0.3.31/src/lib.rs:25:7
   |
25 | #[cfg(feature = "alloc")]
   |       ^^^^^^^^^^^^^^^^^
```

So `async-tungstenite` fails to compile completely.

I will investigate this and fix it as soon as possible.

**Comment by jgraef** [comment]
This has been fixed and released as 0.4.4
